io/disk/iozone.py

```diff
@@ -0,0 +1,577 @@
+class IOzoneAnalyzer(object):
+
+    """
+    Analyze an unprocessed IOzone file, and generate the following types of
+    report:
+
+    * Summary of throughput for all file and record sizes combined
+    * Summary of throughput for all file sizes
+    * Summary of throughput for all record sizes
+
+    If more than one file is provided to the analyzer object, a comparison
+    between the two runs is made, searching for regressions in performance.
+    """
+
+    def __init__(self, list_files, output_dir):
+        self.list_files = list_files
+        if not os.path.isdir(output_dir):
+            os.makedirs(output_dir)
+        self.output_dir = output_dir
+        logging.info("Results will be stored in %s", output_dir)
+
+    @staticmethod
+    def average_performance(results, size=None):
+        """
+        Flattens a list containing performance results.
+        :param results: List of n lists containing data from performance runs.
+        :param size: Numerical value of a size (say, file_size) that was used
+                     to filter the original results list.
+        :return: List with 1 list containing average data from the performance
+                 run.
+        """
+        average_line = []
+        if size is not None:
+            average_line.append(size)
+        for i in range(2, 15):
+            average = data_structures.geometric_mean(
+                [line[i] for line in results]) / 1024.0
+            average = int(average)
+            average_line.append(average)
+        return average_line
+
+    def process_results(self, results, label=None):
+        """
+        Process a list of IOzone results according to label.
+
+        :label: IOzone column label that we'll use to filter and compute
+                geometric mean results, in practical term either 'file_size'
+                or 'record_size'.
+        :result: A list of n x m columns with original iozone results.
+        :return: A list of n-? x (m-1) columns with geometric averages for
+                values of each label (ex, average for all file_sizes).
+        """
+        performance = []
+        if label is not None:
+            index = _LABELS.index(label)
+            sizes = data_structures.ordered_list_unique(
+                [line[index] for line in results])
+            for size in sizes:
+                r_results = [line for line in results if line[index] == size]
+                performance.append(self.average_performance(r_results, size))
+        else:
+            performance.append(self.average_performance(results))
+
+        return performance
+
+    @staticmethod
+    def parse_file(p_file):
+        """
+        Parse an IOzone results file.
+
+        :param file: File object that will be parsed.
+        :return: Matrix containing IOzone results extracted from the file.
+        """
+        lines = []
+        for line in p_file.readlines():
+            fields = line.split()
+            if len(fields) != 15:
```

Comment:
OK, this is going to fail in case of zero-division error as the value is `Error %s/%s`. I don't think we should adjust this code, though. Maybe we could improve the `compare_matrices` to produce splitable output, for example: `error_%s/%s` would work fine. @harish-24 what do you think? Should I send the fix to avocado, or do you have another suggestion?